kernel/comps/mlsdisk/Cargo.toml

```diff
@@ -0,0 +1,22 @@
+[package]
+name = "aster-mlsdisk"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+inherit-methods-macro = {git = "https://github.com/asterinas/inherit-methods-macro", rev = "98f7e3e"}
+ostd-pod = { git = "https://github.com/asterinas/ostd-pod", rev = "c4644be", version = "0.1.1" }
+aster-block = { path = "../block" }
+ostd = { path = "../../../ostd" }
+aes-gcm = { version = "0.9.4", features = ["force-soft"] }
+bittle = "0.5.6"
+ctr = "0.8.0"
```

Comment:
I assume at least some of these crates contain unsafe Rust. Could you provide more information on this topic?

Reply:
The `aes` crate includes some unsafe code, particularly when leveraging high-performance CPU intrinsics like `AES-NI` and `CLMUL`. However, by enabling the `force-soft` feature, we disable these intrinsics, ensuring that the implementation relies solely on software, and in the software implementation, unsafe code is rarely used.

On the other hand, to utilize `AES-NI` and `CLMUL` intrinsics, specific flags must be added to `RUSTFLAGS`, such as `-Ctarget-cpu=sandybridge -Ctarget-feature=+aes,+sse2,+sse4.1,+ssse3`. This requires `Asterinas` to support saving and restoring the `FPU` state within the kernel context, a capability it currently lacks.

Reply:
Thanks for the clarification. So no other crates include unsafe?

Add some comments before the `aes-gcm` crate to explain why using "force-soft".

Reply:
> So no other crates include unsafe?

The `hashbrown` crate has ~300 unsafe, `lru` has ~80, and `postcard` has 14. The rest are all fewer than 10.